Stripe/STPPaymentContext.swift

```diff
@@ -0,0 +1,1129 @@
+  private weak var originalTopViewController: UIViewController?
+  private var applePayVC: PKPaymentAuthorizationViewController?
+
+  // Disable transition animations in tests
+  func transitionAnimationsEnabled() -> Bool {
+    return NSClassFromString("XCTest") == nil
+  }
+
+  var currentValuePromise: STPPromise<STPPaymentOptionTuple> {
+    weak var weakSelf = self
+    return
+      (loadingPromise?.map({ _ in
+        guard let strongSelf = weakSelf, let paymentOptions = strongSelf.paymentOptions else {
+          return STPPaymentOptionTuple()
+        }
+        return STPPaymentOptionTuple(
+          paymentOptions: paymentOptions,
+          selectedPaymentOption: strongSelf.selectedPaymentOption)
+      }))!
+  }
+
+  func remove(_ paymentOptionToRemove: STPPaymentOption?) {
+    // Remove payment method from cached representation
+    var paymentOptions = self.paymentOptions
+    paymentOptions?.removeAll { $0 as AnyObject === paymentOptionToRemove as AnyObject }
+    self.paymentOptions = paymentOptions
+
+    // Elect new selected payment method if needed
+    if let selectedPaymentOption = selectedPaymentOption,
+      selectedPaymentOption.isEqual(paymentOptionToRemove)
+    {
+      self.selectedPaymentOption = self.paymentOptions?.first
+    }
+  }
+
+  // MARK: - Payment Methods
+
+  func presentPaymentOptionsViewController(withNewState state: STPPaymentContextState) {
+    assert(
+      hostViewController != nil && hostViewController?.viewIfLoaded?.window != nil,
+      "hostViewController must not be nil on STPPaymentContext when calling pushPaymentOptionsViewController on it. Next time, set the hostViewController property first!"
+    )
+    if self.state == STPPaymentContextState.none {
+      self.state = state
+      let paymentOptionsViewController = STPPaymentOptionsViewController(paymentContext: self)
+      self.paymentOptionsViewController = paymentOptionsViewController
+      paymentOptionsViewController.prefilledInformation = prefilledInformation
+      paymentOptionsViewController.defaultPaymentMethod = defaultPaymentMethod
+      paymentOptionsViewController.paymentOptionsViewControllerFooterView =
+        paymentOptionsViewControllerFooterView
+      paymentOptionsViewController.addCardViewControllerFooterView = addCardViewControllerFooterView
+      paymentOptionsViewController.navigationItem.largeTitleDisplayMode = largeTitleDisplayMode
+
+      let navigationController = UINavigationController(
+        rootViewController: paymentOptionsViewController)
+      navigationController.navigationBar.stp_theme = theme
+      navigationController.navigationBar.prefersLargeTitles = true
+      navigationController.modalPresentationStyle = modalPresentationStyle
+      hostViewController?.present(
+        navigationController,
+        animated: transitionAnimationsEnabled())
+    }
+  }
+
+  @objc
+  public func paymentOptionsViewController(
```

Comment:
Should not be public

Reply:
Implementations of public protocols need to be public